osschallenge/tests/test_selenium_logged_in_as_contributor.py

```diff
@@ -0,0 +1,142 @@
+    def setUp(self):
+        self.client = Client()
+        self.login_page = LoginPage(self.driver, self.live_server_url)
+        self.register_page = RegisterPage(self.driver, self.live_server_url)
+        self.new_project_page = NewProjectPage(
+            self.driver, self.live_server_url
+        )
+        self.new_task_page = NewTaskPage(self.driver, self.live_server_url)
+
+        self.user1 = User.objects.create(
+            last_login="2017-10-18 11:55:45.681893+00",
+            is_superuser=False,
+            username="Test",
+            first_name="Test",
+            last_name="Test",
+            email="example@example.ch",
+            is_staff=False,
+            is_active=True,
+            date_joined="2017-10-13 08:17:36.901715+00"
+        )
+        self.user1.set_password("12345qwert")
+        self.user1.save()
+
+        self.login_page.open()
+        self.login_page.login("Test", "12345qwert")
+
+        self.role1 = Role.objects.create(
+            id=1,
+            name="Contributor"
+        )
+
+        self.profile1 = Profile.objects.create(
+            user=self.user1,
+            role=self.role1,
+            links="Test",
+            contact="Test",
+            key="Test1",
+            picture="Test.png"
+        )
+
+        self.project1 = Project.objects.create(
+            id=1,
+            title_de="OpenStreetMap",
+            title_en_us="OpenStreetMap",
+            lead_text_de="Blablablab",
+            lead_text_en_us="Blablablab",
+            description_de="Blablablab",
+            description_en_us="Blablablab",
+            licence="MIT",
+            website="www.google.ch",
+            github="www.github.com",
+            owner=self.user1
+        )
+
+        self.task1 = Task.objects.create(
+            id=1,
+            title="Bug Fixing",
+            lead_text="Bug Fixing",
+            description="Bug Fixing",
+            project=self.project1,
+            assignee=None,
+            task_done=False,
+            task_checked=False,
+            picture="test.png",
+            approved_by=None,
+            approval_date="2017-10-18 12:34:51.168157+00"
+        )
+
+        self.client.login(username="Test", password='12345qwert')
+        cookie = self.client.cookies['sessionid']
+        self.driver.get(self.live_server_url + '/login/')
+        self.driver.add_cookie(
+            {
+                'name': 'sessionid',
+                'value': cookie.value,
+                'secure': False,
+                'path': '/'
+            }
+        )
+        self.driver.refresh()
+        self.driver.get(self.live_server_url + '/login/')
+
+    @classmethod
+    def tearDownClass(self):
+        self.driver.quit()
+        super(MydriverTests, self).tearDownClass()
+
+    def test_write_a_comment(self):
+        self.driver.get("{}{}".format(self.live_server_url, "/tasks/1/"))
+        comment_input = self.driver.find_element_by_id("markdown-comment")
```

Comment:
If you want to apply the page object pattern 100%, it's not allowed to have calls like `find_element_..` in the test itself anymore - those all have to be in page objects. It might make sense to create one for this case, too?

Reply:
All code is now outsourced in page objects.